Common/Python/PandasData.cs

```diff
@@ -0,0 +1,280 @@
+        /// </summary>
+        /// <param name="baseData"><see cref="IBaseData"/> object that contains information to be saved in an instance of <see cref="PandasData"/></param>
+        public void Add(IBaseData baseData)
+        {
+            _timeIndex.Add(baseData.EndTime);
+
+            var bar = baseData as IBaseDataBar;
+            if (bar != null)
+            {
+                _series["open"].Add((double)bar.Open);
+                _series["high"].Add((double)bar.High);
+                _series["low"].Add((double)bar.Low);
+                _series["close"].Add((double)bar.Close);
+
+                var tradeBar = bar as TradeBar;
+                if (tradeBar != null)
+                {
+                    _series["volume"].Add((double)tradeBar.Volume);
+                }
+
+                var quoteBar = bar as QuoteBar;
+                if (quoteBar != null)
+                {
+                    _series["asksize"].Add((double)quoteBar.LastAskSize);
+                    _series["bidsize"].Add((double)quoteBar.LastBidSize);
+
+                    if (quoteBar.Ask != null)
+                    {
+                        _series["askopen"].Add((double)quoteBar.Ask.Open);
+                        _series["askhigh"].Add((double)quoteBar.Ask.High);
+                        _series["asklow"].Add((double)quoteBar.Ask.Low);
+                        _series["askclose"].Add((double)quoteBar.Ask.Close);
+                    }
+                    else
+                    {
+                        _series["askopen"].Add(double.NaN);
+                        _series["askhigh"].Add(double.NaN);
+                        _series["asklow"].Add(double.NaN);
+                        _series["askclose"].Add(double.NaN);
+                    }
+
+                    if (quoteBar.Bid != null)
+                    {
+                        _series["bidopen"].Add((double)quoteBar.Bid.Open);
+                        _series["bidhigh"].Add((double)quoteBar.Bid.High);
+                        _series["bidlow"].Add((double)quoteBar.Bid.Low);
+                        _series["bidclose"].Add((double)quoteBar.Bid.Close);
+                    }
+                    else
+                    {
+                        _series["bidopen"].Add(double.NaN);
+                        _series["bidhigh"].Add(double.NaN);
+                        _series["bidlow"].Add(double.NaN);
+                        _series["bidclose"].Add(double.NaN);
+                    }
+                }
+            }
+
+            var tick = baseData as Tick;
+            if (tick != null)
+            {
+                if (tick.TickType == TickType.Quote)
+                {
+                    _series["askprice"].Add((double)tick.AskPrice);
+                    _series["asksize"].Add((double)tick.AskSize);
+                    _series["bidprice"].Add((double)tick.BidPrice);
+                    _series["bidsize"].Add((double)tick.BidSize);
```

Comment:
not sure if we can handle this, but sometimes ticks only have bid or ask values, so may end up with zeros here.